tools/src/wyvern/tools/parsing/coreparser/ASTBuilder.java

```diff
@@ -13,7 +13,7 @@
 	/** if type is null, it will be inferred*/
 	public AST valDecl(String name, Type type, AST exp, FileLocation loc);
 	public AST varDecl(String name, Type type, AST exp, FileLocation loc);
-	public AST defDecl(String name, Type type, List args, AST body, boolean isClassDef, FileLocation loc);
+	public AST defDecl(String name, Type type, List<Token> generics, List args, AST body, boolean isClassDef, FileLocation loc);
```

Comment:
I feel the same way, but I found other examples of tokens in the AST so I figured it was OK.
I can remove this now that I have a `GenericList` production and the parsing code is structured more logically.